// @chakra-ui/cli/dist/utils/init-cli.js
"use strict";

exports.__esModule = true;
exports.initCLI = initCLI;

var _cliCheckNode = _interopRequireDefault(require("cli-check-node"));

var _cliWelcome = _interopRequireDefault(require("cli-welcome"));

var _updateNotifier = _interopRequireDefault(require("update-notifier"));

var _cliHandleUnhandled = _interopRequireDefault(require("cli-handle-unhandled"));

var _package = _interopRequireDefault(require("../../package.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function initCLI() {
  return _initCLI.apply(this, arguments);
}

function _initCLI() {
  _initCLI = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
    return regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            (0, _cliCheckNode["default"])("12");
            _context.next = 3;
            return (0, _cliHandleUnhandled["default"])();

          case 3:
            (0, _cliWelcome["default"])({
              title: "Chakra UI CLI",
              tagLine: "by Chakra UI\n" + _package["default"].description,
              bgColor: "#319795",
              color: "#FFFFFF",
              bold: true,
              clear: false,
              version: _package["default"].version
            });
            (0, _updateNotifier["default"])({
              pkg: _package["default"],
              shouldNotifyInNpmScript: true,
              updateCheckInterval: 1000 * 60 * 60 * 24 * 3 // 3 days

            }).notify({
              isGlobal: true
            });

          case 5:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));
  return _initCLI.apply(this, arguments);
}
